addons/mail/controllers/discuss.py

```diff
@@ -198,6 +198,14 @@ def mail_init_messaging(self, **kwargs):
             return guest.sudo()._init_messaging()
         raise NotFound()
 
+    @http.route('/mail/channel/members', methods=['POST'], type='json', auth='public')
+    def mail_channel_members(self, **kwargs):
+        request.env['mail.channel.member']._get_as_sudo_from_request_or_raise(request=request, channel_id=kwargs['channel_id'])
+        channel_sudo = request.env['mail.channel'].browse(kwargs['channel_id']).sudo().exists()
+        if channel_sudo:
+            return channel_sudo.load_more_members(kwargs['known_member_ids'])
+        raise NotFound()
```

Comment:
When `_get_as_sudo_from_request_or_raise` does not raise, the channel inevitably exists. So could just `sudo` the `load_more_members`